pandas/core/indexes/base.py

```diff
@@ -5474,6 +5484,14 @@ def _should_compare(self, other: Index) -> bool:
 
         other = unpack_nested_dtype(other)
         dtype = other.dtype
+        if other._is_multi:
+            if not self._is_multi:
+                # other contains only tuples so unless we are object-dtype,
+                #  there can never be any matches
+                return self._is_comparable_dtype(dtype)
+            return self.nlevels == other.nlevels
```

Comment:
This is the change that is breaking MultiIndex broadcasting.  If one has 3 levels and the other has 2, then this is False.  Previously these were comparable and so would be compared and expanded.

Reply:
thanks.  do you know what the calling method is in the problematic case?

Reply:
Walking back, previous is

https://github.com/pandas-dev/pandas/blob/ddd90b0f171fac3c2e39bc2ff76df97d57f36a18/pandas/core/indexes/base.py#L3481

then 

https://github.com/pandas-dev/pandas/blob/ddd90b0f171fac3c2e39bc2ff76df97d57f36a18/pandas/core/indexes/base.py#L3887

Here `self` is the Series with 2 levels and other is the DataFrame with 3.

Reply:
The big change is driven by the return difference of `self._should_compare(target)`. Before this patch it returned `True`, so the `if not ...` block was skipped.  It now returns `False`, and so it incorrectly shortcuts and fills with an NA value.

Reply:
OK, i think ive got a handle on whats going on here.  The long-term fix will be in MultiIndex.get_indexer, but for now this should just be reverted.